sdk/src/main/java/com/bugsnag/android/DeviceData.java

```diff
@@ -70,46 +83,115 @@ public void toStream(@NonNull JsonStream writer) throws IOException {
 
         writer
             .name("id").value(id)
-            .name("freeMemory").value(getFreeMemory())
-            .name("totalMemory").value(getTotalMemory())
-            .name("freeDisk").value(getFreeDisk())
-            .name("orientation").value(getOrientation(appContext));
-
-
-        // TODO migrate metadata values
-
-        writer
-            .name("batteryLevel").value(getBatteryLevel(appContext))
-            .name("charging").value(isCharging(appContext))
-            .name("locationStatus").value(getLocationStatus(appContext))
-            .name("networkAccess").value(getNetworkAccess(appContext))
-            .name("time").value(getTime())
-            .name("brand").value(Build.BRAND)
-            .name("apiLevel").value(Build.VERSION.SDK_INT)
-            .name("osBuild").value(Build.DISPLAY)
-            .name("locale").value(locale)
-            .name("screenDensity").value(screenDensity)
-            .name("dpi").value(dpi)
-            .name("emulator").value(isEmulator())
-            .name("screenResolution").value(screenResolution);
-
-        writer.name("cpuAbi").beginArray();
-        for (String s : cpuAbi) {
-            writer.value(s);
-        }
-        writer.endArray();
+            .name("freeMemory").value(freeMemory)
+            .name("totalMemory").value(totalMemory)
+            .name("freeDisk").value(freeDisk)
+            .name("orientation").value(orientation);
         writer.endObject();
     }
 
-    @NonNull
-    String getUserId() {
+    // TODO migrate metadata values to separate class
+    void addDeviceMetaData(MetaData metaData) {
+        metaData.addToTab("device", "batteryLevel", getBatteryLevel(appContext));
+        metaData.addToTab("device", "charging", isCharging(appContext));
+        metaData.addToTab("device", "locationStatus", getLocationStatus(appContext));
+        metaData.addToTab("device", "networkAccess", getNetworkAccess(appContext));
+        metaData.addToTab("device", "time", getTime());
+        metaData.addToTab("device", "brand", Build.BRAND);
+        metaData.addToTab("device", "apiLevel", Build.VERSION.SDK_INT);
+        metaData.addToTab("device", "osBuild", Build.DISPLAY);
+        metaData.addToTab("device", "locale", locale);
+        metaData.addToTab("device", "screenDensity", screenDensity);
+        metaData.addToTab("device", "dpi", dpi);
+        metaData.addToTab("device", "emulator", isEmulator());
+        metaData.addToTab("device", "screenResolution", screenResolution);
+        metaData.addToTab("device", "cpuAbi", cpuAbi);
+    }
```

Comment:
How were you thinking that the unity notifier would access all of these properties?

Reply:
For the Android notifier, these need to be added to `error.metadata`. My thinking is that we could expose an internal method on `Client` that allows for the population of this metadata. Something like:

```